tests/Application.UnitTests/Common/EntityRequestValidators/CreateAddressRequestValidatorTests.cs

```diff
@@ -0,0 +1,228 @@
+
+        result.Errors.Where(e => e.PropertyName.Equals(nameof(CreateAddressRequest.PostalCode)))
+            .Should().ContainEquivalentOf(expectedFailure, cfg => cfg.Excluding(e => e.FormattedMessagePlaceholderValues));
+    }
+
+    [Test]
+    public void State_ValidInput_NoValidationFailures()
+    {
+        _createAddressRequest.State = "Valid State";
+
+        var result = _createAddressRequestValidator.Validate(_createAddressRequest);
+
+        result.Errors.Where(e => e.PropertyName.Equals(nameof(CreateAddressRequest.State)))
+            .Should().BeEmpty();
+    }
+
+    [Test]
+    public void State_IsNull_NoValidationFailure()
+    {
+        _createAddressRequest.State = null;
+
+        var result = _createAddressRequestValidator.Validate(_createAddressRequest);
+
+        result.Errors.Where(e => e.PropertyName.Equals(nameof(CreateAddressRequest.State)))
+            .Should().BeEmpty();
+    }
+
+    [Test]
+    public void State_ExceedsMaxLength_ValidationFailure()
+    {
+        _createAddressRequest.State = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
+
+        var expectedFailure = new ValidationFailure()
+        {
+            PropertyName = nameof(CreateAddressRequest.State),
+            AttemptedValue = _createAddressRequest.State,
+            ErrorMessage = "The length of 'State' must be 100 characters or fewer. You entered 101 characters.",
+            ErrorCode = "MaximumLengthValidator"
+        };
+
+        var result = _createAddressRequestValidator.Validate(_createAddressRequest);
+
+        result.Errors.Where(e => e.PropertyName.Equals(nameof(CreateAddressRequest.State)))
+            .Should().ContainEquivalentOf(expectedFailure, cfg => cfg.Excluding(e => e.FormattedMessagePlaceholderValues));
+    }
+
+    [Test]
+    public void StreetAddress_ValidInput_NoValidationFailures()
+    {
+        _createAddressRequest.StreetAddress = "Valid StreetAddress";
+
+        var result = _createAddressRequestValidator.Validate(_createAddressRequest);
+
+        result.Errors.Where(e => e.PropertyName.Equals(nameof(CreateAddressRequest.StreetAddress)))
+            .Should().BeEmpty();
+    }
+
+    [Test]
+    public void StreetAddress_IsNull_NoValidationFailure()
+    {
+        _createAddressRequest.StreetAddress = null;
+
+        var result = _createAddressRequestValidator.Validate(_createAddressRequest);
+
+        result.Errors.Where(e => e.PropertyName.Equals(nameof(CreateAddressRequest.StreetAddress)))
+            .Should().BeEmpty();
+    }
+
+    [Test]
+    public void StreetAddress_ExceedsMaxLength_ValidationFailure()
+    {
+        _createAddressRequest.StreetAddress = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
```

Comment:
_createAddressRequest.StreetAddress = new string('a', 300);

I would say this is a better way to do it but you can leave this implementation for now :)